### PR TITLE
mod_base: set the CSP for files to: default-src 'none'; sandbox

### PR DESCRIPTION
**Pull request by mworrell**
### Description

This makes displaying user uploaded content even more restrictive than the previous `sandbox` solution.

### Checklist

- [ ] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

**Comment by mworrell** [comment]
See also #3108

File: apps/zotonic_mod_base/src/controllers/controller_file.erl [hunk 1]
```diff
@@ -217,14 +217,18 @@ set_content_policy(#z_file_info{ mime = Mime } = Info, Context) ->
     case is_resource(Info) of
         true when Mime =:= <<"application/pdf">> ->
             z_context:set_resp_headers([
-                    {<<"content-security-policy">>, <<"object-src 'self'; plugin-types application/pdf">>},
-                    {<<"x-content-security-policy">>,<<"plugin-types: application/pdf">>}
+                    {<<"content-security-policy">>, <<"default-src 'none'; object-src 'self'; plugin-types application/pdf">>},
+                    {<<"x-content-security-policy">>,<<"default-src 'none'; plugin-types application/pdf">>}
```
[the remark this answers is not in the document]
Comment (robvandenbogaard) [reply apps/zotonic_mod_base/src/controllers/controller_file.erl:221]: I forgot this is in controller_file and not site-wide, seems ok to set it like this. Though looking up `plugin-types` I read that that policy is deprecated. Is there a specific reason to include it anyway? See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/plugin-types

[the remark this answers is not in the document]
Comment (mworrell) [reply apps/zotonic_mod_base/src/controllers/controller_file.erl:221]: Maybe for Safari?

This code is there quite some time already, maybe from before the deprecation.

Is there anything else needed for application/pdf or could we just remove it?

[the remark this answers is not in the document]
Comment (mworrell) [reply apps/zotonic_mod_base/src/controllers/controller_file.erl:221]: And indeed, this is only for user-uploaded files. Anything else is served with the default CSP header for the site.

**Review by robvandenbogaard — APPROVED** [review]

**Pull request merged**